### PR TITLE
Improve the pkg

### PR DESCRIPTION
**Pull request by 3imed-jaberi**
- update the `.gitignore` file.
- avoid `package-lock.json` file.
- update all dependencies.
- update the pipeline (node 8.x).
- improve the pkg by replace the full loadash dep. with only needed funcs.

> can go to next version 🚀 !

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x, 12.x]
+        node-version: [8.x, 10.x, 12.x]
```
Comment (jonathanong) [inline .github/workflows/node.js.yml:12]: node 8 is no longer supported

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -21,7 +21,6 @@ jobs:
       uses: actions/setup-node@v1
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm ci
-    - run: npm run eslint
-    - run: npm run test -- --coverage
+    - run: npm install
+    - run: npm run ci
```
Comment (jonathanong) [inline .github/workflows/node.js.yml:25]: don't want to combine the jobs because it will make debugging tests harder

File: .npmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (jonathanong) [inline .npmrc:1]: no reason to remove it

**Comment by 3imed-jaberi** [comment]
I think you did this improvement yourself .. so I don’t see why you should leave this open.

**Pull request closed without merging**